### PR TITLE
Feature: support mail as a provider

### PR DESCRIPTION
**Pull request by Andrew44Ashraf**
Fixes # 27.

Changes proposed in this pull request:
-support email as a provider

### CODE REVIEW

**Review by amgadramses — APPROVED** [review]

File: provider/email.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	host     string
+	port     int
+	to       string
+}
+
+// NewEmail returns new email instance
+func NewEmail(from string, password string, host string, port int, to string) Provider {
+	if len(from) == 0 {
+		logrus.Warnf("initializing email with an empty from")
+	} else if len(password) == 0 {
```
Comment (abahmed) [inline provider/email.go:24]: I think we don't need else here to detect all missing values

File: util/util.go [hunk 2]
```diff
@@ -117,10 +118,32 @@ func GetProviders() []provider.Provider {
 			if key == "telegram" && c == "chatid" && len(strings.TrimSpace(v.(string))) > 0 {
 				telegram[1] = true
 			}
+			if key == "email" && c == "from" && len(strings.TrimSpace(v.(string))) > 0 {
```
Comment (abahmed) [inline util/util.go:121]: I think there's a better way (more readable) for this validation

**Review by abahmed — CHANGES_REQUESTED** [review]

File: util/util.go (hunk 2 above)
Comment (Andrew44Ashraf) [reply util/util.go:121]: @abahmed will be refactored in another PR

File: provider/email.go (hunk 1 above)
Comment (Andrew44Ashraf) [reply provider/email.go]: updated

**Pull request merged**